src/cli/self_update/shell.rs

```diff
@@ -445,6 +446,62 @@ impl UnixShell for Pwsh {
     }
 }
 
+struct Xonsh;
+
+impl UnixShell for Xonsh {
+    fn does_exist(&self, process: &Process) -> bool {
+        process.var("XONSHRC").is_ok() || utils::find_cmd(&["xonsh"], process).is_some()
+    }
+
+    fn rcfiles(&self, process: &Process) -> Vec<PathBuf> {
+        let mut paths = vec![];
+
+        if let Ok(p) = process.var("XDG_CONFIG_HOME") {
+            let mut p = PathBuf::from(p);
+            p.extend(["xonsh", "rc.xsh"]);
+            paths.push(p);
+        }
+
+        if let Some(mut p) = process.home_dir() {
+            p.extend([".config", "xonsh", "rc.xsh"]);
+            paths.push(p);
+        }
+
+        if let Some(home) = process.home_dir() {
+            paths.push(home.join(".xonshrc"));
+        }
+
+        paths
+    }
+
+    fn update_rcs(&self, process: &Process) -> Vec<PathBuf> {
+        for f in self.rcfiles(process) {
+            if f.is_file() {
```

Comment:
Please don't copy from `tcsh` here, it was written in this way because both `.tcshrc` and `.cshrc` are supported as valid `tcsh` RC files.

Copying the implementation from a more regular shell like `Nu` would be fine.

Reply:
@rami3l could you please change this part of code as you prefer to see? The logic looks trivial but I will spend much time on understanding what you want to see than really doing implementation. If you have no time I'm ok to close this until someone will grab this.

Reply:
Updated and [tested](https://github.com/rust-lang/rustup/pull/4626#issue-3674255563).